etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/EtorDomainRegistration.java

```diff
@@ -126,9 +126,19 @@ DomainResponse handleDemographics(DomainRequest request) {
     DomainResponse handleOrders(DomainRequest request) {
         Order<?> orders;
 
+        String submissionId = request.getHeaders().get("RecordId");
+        // TODO: need to verify what we should do if we don't receive the RecordId header or if it's
+        // empty
+        // For now, returning a 400 error
+        if (submissionId == null) {
```

Comment:
Need to discuss what to do when we don't receive the RecordId

Reply:
After discussion with the team, decided not break the flow and log it as an error